src/state.rs

```diff
@@ -337,4 +361,19 @@ mod test {
         let (_, in_check_state_2) = fen("8/8/8/8/8/pk6/P1p5/1KP5 b").unwrap();
         assert!(in_check_state_2.in_check());
     }
+
+    #[test]
+    fn test_two_square_advance() {
+        let (_, initial_state) = fen("rnbqkbnr/pppppppp/8/8/8/8/PPPPPPPP/RNBQKBNR w").unwrap();
```

Comment:
this way of representing a board is really cool!

Reply:
i know! it makes testing much better than it used to be: https://github.com/mjhoy/chess/blob/7309b9f3fdb7105651223e23971dda505b1880ef/src/state.rs#L294